packages/next-preact/index.js

```diff
@@ -18,6 +21,77 @@ module.exports = (nextConfig = {}) => {
         'react-dom': 'preact-compat'
       })
 
+      const commonsPlugin = new webpack.optimize.CommonsChunkPlugin({
+        name: 'main.js',
+        filename: options.dev
+          ? 'static/commons/main.js'
+          : 'static/commons/main-[chunkhash].js',
+        minChunks(module, count) {
+          // React and React DOM are used everywhere in Next.js. So they should always be common. Even in development mode, to speed up compilation.
+          if (
+            module.resource &&
+            module.resource.includes(`${sep}react-dom${sep}`) &&
+            count >= 0
+          ) {
+            return true
+          }
+
+          if (
+            module.resource &&
+            module.resource.includes(`${sep}react${sep}`) &&
+            count >= 0
+          ) {
+            return true
+          }
+
+          if (
+            module.resource &&
+            module.resource.includes(`${sep}preact-compat${sep}`) &&
+            count >= 0
+          ) {
+            return true
+          }
+
+          if (
+            module.resource &&
+            module.resource.includes(`${sep}preact${sep}`) &&
+            count >= 0
+          ) {
+            return true
+          }
+
+          // In the dev we use on-demand-entries.
+          // So, it makes no sense to use commonChunks based on the minChunks count.
+          // Instead, we move all the code in node_modules into each of the pages.
+          if (options.dev) {
+            return false
+          }
+
+          // Commons
+          // If there are one or two pages, only move modules to common if they are
+          // used in all of the pages. Otherwise, move modules used in at-least
+          // 1/2 of the total pages into commons.
+          if (options.totalPages <= 2) {
+            return count >= options.totalPages
+          }
+          return count >= options.totalPages * 0.5
+          // Commons end
+        }
+      })
+
+      const plugins = options.isServer
+        ? config.plugins
+        : // Overrides common chunk plugin including preact
+          config.plugins.map(plugin => {
+            if (plugin.minChunks instanceof Function) {
```

Comment:
This specific line, for example, it's not very trustworthy IMO
we could end up overriding plugins which we don't mean to with such code. 😕